### PR TITLE
Add cli cmd to list the provider trigger info

### PR DESCRIPTION
**Pull request by pankajastro**
related to: https://github.com/apache/airflow/issues/29918

Add cli cmd `airflow providers triggers` this command list the table with the column provider_package.name, provider trigger class name and integration name 
example
Screenshot 2023-04-23 at 1 05 52 AM

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by pankajastro** [comment]
Hi @potiuk / @eladkal need some feedback on this. thank you!

**Review by potiuk — APPROVED** [review]
Makes perfect sense!

File: airflow/providers_manager.py [hunk 1]
```diff
@@ -941,6 +950,27 @@ def _discover_auth_backends(self) -> None:
                     if _sanity_check(provider_package, auth_backend_module_name + ".init_app", provider):
                         self._api_auth_backend_module_names.add(auth_backend_module_name)
 
+    @provider_info_cache("triggers")
+    def initialize_providers_triggers(self):
+        """initialization of providers triggers."""
```
Comment (potiuk) [inline airflow/providers_manager.py]:
```suggestion
        """Initialization of providers triggers."""
```

**Comment by potiuk** [comment]
Just the static check failure with capitalization

File: airflow/cli/cli_config.py [hunk 2]
```diff
@@ -1754,6 +1754,12 @@ class GroupCommand(NamedTuple):
         func=lazy_load_command("airflow.cli.commands.provider_command.hooks_list"),
         args=(ARG_OUTPUT, ARG_VERBOSE),
     ),
+    ActionCommand(
+        name="triggers",
+        help="List registered provider triggers",
+        func=lazy_load_command("airflow.cli.commands.provider_command.triggers_list"),
+        args=(ARG_OUTPUT, ARG_VERBOSE),
```
Comment (pankajkoti) [inline airflow/cli/cli_config.py:1761]: Would it make sense to provide an additional optional argument for provider names to filter the triggers by provider name as the list could grow long?

Comment (potiuk) [reply airflow/cli/cli_config.py:1761]: If we do, we should add it for all other provider commands in a separate PR - this should be consistent across all providers CLI commands.

Comment (pankajkoti) [reply airflow/cli/cli_config.py:1761]: okay, thanks. I will take that up in a separate PR.

**Pull request merged**